packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -222,14 +224,74 @@ const duplicateNameError = async (context: HookContext<LocationService>) => {
   }
 }
 
+/* BEFORE FIND HOOKS */
+/**
+ * @function rejectEmptyQuery
+ * @param context
+ * @description This hook function ensure that no empty queries are present
+ */
+const rejectEmptyQuery = async (context: HookContext<LocationService>) => {
+  const keys: string[] = Object.keys(context.params.query || {})
+  if (keys.length == 0) throw new BadRequest(`Query can't have empty params object`)
+}
+/**
+ * @function restrictProtectedQuery
+ * @param context
+ * @description This hook function restricts allowed query params for scoped user's calls
+ * from authorized sections (admin, dashboard, studio)
+ */
+const restrictProtectedQuery = async (context: HookContext<LocationService>) => {
+  keepQuery(
+    'id',
+    'name',
+    'slugifiedName',
+    '$like',
+    '$limit',
+    '$sort',
+    '$select',
+    '$skip',
+    'sceneId',
+    'projectId'
+  )(context)
+}
+/**
+ * @function restrictPublicQuery
+ * @param context
+ * @description This hook function restricts allowed query params for public calls
+ * from the viewer section by only allowing the slugifiedName parameter, any other
+ * parameter will be discarded from the query protecting from any malicious query injection.
+ *
+ * This function is discarding all params but slugifiedName instead of throwing a BadRequest if other
+ * parameters are present, having as goal to avoid providing any feedback to malicious users
+ * that could allow them to figure out unforeseen and unsafe patterns or combinations in order to
+ * successfully perform an injection attack.
+ */
+const restrictPublicQuery = async (context: HookContext<LocationService>) => {
+  keepQuery('slugifiedName')(context)
+}
+
 export default {
   around: {
     all: [schemaHooks.resolveExternal(locationExternalResolver), schemaHooks.resolveResult(locationResolver)]
   },
 
   before: {
     all: [schemaHooks.validateQuery(locationQueryValidator), schemaHooks.resolveQuery(locationQueryResolver)],
-    find: [discardQuery('action'), discardQuery('studio'), sortByLocationSetting],
+    find: [
+      // Empty query object in findLocation is not allowed
+      rejectEmptyQuery,
+      // Persist query to avoid hook collitions or malformed knex query instances
+      persistQuery,
+      // Clean up query of non db relevant parameters
+      discardQuery('action'),
+      discardQuery('studio'),
```

Comment:
```suggestion
```
I think we can remove this `discardQuery('studio'),` as I dont see it used anywhere in code.